app/services/submission_service.rb

```diff
@@ -327,5 +327,43 @@ def add_user_to_submission(submission, gravity_user_id, access_token)
 
       submission
     end
+
+    def list_artwork(submission, gravity_partner_id, access_token)
+      raise SubmissionError, "Partner is required" unless gravity_partner_id.present?
+      # Validate artist is set and public
+      raise SubmissionError, "Must have an artist to list" if submission.artist_id.blank?
+      artist = GravityV1.get("/api/v1/artist/#{submission.artist_id}")
+      raise SubmissionError, "Artist must be public" unless artist["public"]
+
+      # TODO: Source data from Salesforce or MyCollection artworks as desired
+
+      # Create artwork
+      artwork = GravityV1.post("/api/v1/artwork",
+        params: submission.to_artwork_params.merge(partner: gravity_partner_id),
+        token: access_token)
+
+      # Add edition set if applicable
+      if submission.edition?
+        GravityV1.post("/api/v1/artwork/#{artwork["_id"]}/edition_set",
+          params: submission.to_edition_set_params,
+          token: access_token)
+      end
+
+      # Add images, primary image first
+      submission.assets
+        .sort_by.with_index { |a, i| submission.primary_image && a == submission.primary_image ? -1 : i }
+        .map { |a| a.original_image }
+        .each do |url|
+        GravityV1.post("/api/v1/artwork/#{artwork["_id"]}/image",
+          params: {remote_image_url: url, low_priority: true},
+          token: access_token)
+      end
+
+      # Record newly listed artwork
+      submission.listed_artwork_ids << artwork["_id"]
+      submission.save!
```

Comment:
**question**: For a My Collection artwork in Gravity that's associated with this submission (via `submission_id=[ID]`) do we need to do anything else to also add this listed artwork ID to its `listed_artwork_ids` field?

Reply:
Yes good point. We shouldn't forget the whole point of this exercise: product support and data continuity across the whole consignment lifecycle.

I figured we'd update Gravity to set that relationship when a new artwork is created this way (the listed artwork's `external_id` may match a My Collection work's `submission_id`). I think that's fine, and matches how Salesforce-imported works operate, but we might also try to push the update from here. What do you think?

Reply:
Yeah, I think that makes sense as a Gravity change. No need to complicate the operation here.